src/platform_impl/web/event_loop/proxy.rs

```diff
@@ -1,17 +1,111 @@
-use super::runner::WeakShared;
-use crate::platform_impl::platform::r#async::Waker;
+use std::future;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::Arc;
+use std::task::Poll;
 
-#[derive(Clone)]
-pub struct EventLoopProxy {
-    runner: Waker<WeakShared>,
+use super::super::main_thread::MainThreadMarker;
+use crate::event_loop::EventLoopProxyProvider;
+use crate::platform_impl::web::event_loop::runner::WeakShared;
+use crate::platform_impl::web::r#async::{AtomicWaker, Wrapper};
+
+pub struct EventLoopProxy(Wrapper<Handler, Sender, ()>);
+
+struct Handler {
+    execution: WeakShared,
+    handler: fn(&WeakShared, bool),
 }
 
+#[derive(Clone)]
+struct Sender(Arc<Inner>);
+
 impl EventLoopProxy {
-    pub fn new(runner: Waker<WeakShared>) -> Self {
-        Self { runner }
+    pub fn new(
+        main_thread: MainThreadMarker,
+        execution: WeakShared,
+        handler: fn(&WeakShared, bool),
```

Comment:
I think we can remove this parameter here as well and just inline the code.

Can do it myself as well tonight.

Reply:
will leave it up to you.

Reply:
Done.